### ISSUE TITLE
TBC: Errors opening Interface Options / Cannot access AddOn options

### ISSUE THREAD

**Issue by SDPhantom** [Stanzilla/WoWUIBugs#77]
Upon opening Interface Options, a series of errors pop up indicating textures that have been relocated or removed.
Build# 2.5.1.38169

&nbsp;

Opening Interface Options:
> Interface\FrameXML\InterfaceOptionsFrame.lua:158: attempt to index global 'InterfaceOptionsFrameCategoriesTop' (a nil value)
> Interface\FrameXML\InterfaceOptionsFrame.lua:159: attempt to index global 'InterfaceOptionsFrameAddOnsTop' (a nil value)

Both of these textures have been converted to using the Backdrop mixin, which is based off the NineSliceUtils. They have been relocated to the following objects respectively.
`InterfaceOptionsFrameCategories.TopEdge`
`InterfaceOptionsFrameAddOns.TopEdge`

&nbsp;

Clicking the AddOns Tab:
> Interface\FrameXML\InterfaceOptionsFrame.lua:423: attempt to index global 'InterfaceOptionsFrameTab1TabSpacer' (a nil value)
> Interface\FrameXML\InterfaceOptionsFrame.lua:424: attempt to index global 'InterfaceOptionsFrameTab2TabSpacer1' (a nil value)
> Interface\FrameXML\InterfaceOptionsFrame.lua:425: attempt to index global 'InterfaceOptionsFrameTab2TabSpacer2' (a nil value)

These textures have been removed entirely.

&nbsp;

Selecting an AddOn Category:
> Interface\FrameXML\OptionsFrameTemplates.lua:96: attempt to index local 'panelContainer' (a nil value)

I received this in testing my own addons, but other errors may occur. It took me a long time to track this down. This happens when an options panel registered by an addon doesn't have its parent set. According to the currently running code, options panels are expected to be parented to `InterfaceOptionsFramePanelContainer`. A quick fix would be to enforce this when an addon calls `InterfaceOptions_AddCategory()`, perhaps setting the parent itself.

**Comment by Meorawr** [comment]
Attaching a minimal example to assist with reproducing all the above.

```lua
-- Bug #1: Opening the Interface Options panel fails; uncommenting the below fixes it
-- just enough to allow testing of the other issues.

--InterfaceOptionsFrameCategoriesTop = InterfaceOptionsFrameCategories.TopEdge;
--InterfaceOptionsFrameAddOnsTop = InterfaceOptionsFrameAddOns.TopEdge;

-- Bug #2: Clicking the red "+" arrow on for the "Parent" entry this creates
-- in the AddOns tab will trigger errors and fail to expand the sublist.

local Panel = CreateFrame("Frame", "MyAddonPanel", UIParent);
Panel.name = "Parent";
InterfaceOptions_AddCategory(Panel);

local Child = CreateFrame("Frame", "MyAddonChild", Panel);
Child.name = "Child";
Child.parent = Panel.name;
InterfaceOptions_AddCategory(Child);

-- Bug #3: Clicking the "Parentless" entry in the AddOns tab will trigger
-- errors on indexing the (nil) parentContainer.

local Parentless = CreateFrame("Frame", "MyParentlessPanel", nil);
Parentless.name = "Parentless";
InterfaceOptions_AddCategory(Parentless);
```

**Comment by Meorawr** [comment]
Bugs 1 and 2 should be fixed in 2.5.1.38285, bug 3 is a wontfix though from what I recall as it's not really a bug and more just improper use of the API.
